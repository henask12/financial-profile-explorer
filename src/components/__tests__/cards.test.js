import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CompanyCard from '../card/companiesCard';

// Mocking react-router-dom's useNavigate
jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

describe('CompanyCard Component', () => {
  const mockCompany = {
    image: 'https://financialmodelingprep.com/image-stock/PYPL.png',
    companyName: 'PayPal Holdings, Inc.',
    symbol: 'PYPL',
    price: 62.02,
  };

  it('renders the CompanyCard component correctly', () => {
    const { getByText, getByAltText } = render(<CompanyCard company={mockCompany} index={0} />);
    
    expect(getByText(mockCompany.symbol)).toBeInTheDocument();
    expect(getByText(mockCompany.price)).toBeInTheDocument();
    expect(getByAltText(mockCompany.companyName)).toBeInTheDocument();
  });

  it('calls navigate when the button is clicked', () => {
    const mockNavigate = jest.fn();
    require('react-router-dom').useNavigate.mockReturnValue(mockNavigate);
    
    const { getByRole } = render(<CompanyCard company={mockCompany} index={0} />);
    const button = getByRole('button');
    
    fireEvent.click(button);
    expect(mockNavigate).toHaveBeenCalledWith(`/company/${mockCompany.symbol}`, {
      state: { company: mockCompany },
    });
  });
});

