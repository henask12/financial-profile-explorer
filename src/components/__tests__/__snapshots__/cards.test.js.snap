// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard correctly validates PropTypes 1`] = `
<div>
  <button
    class="border-0 bg-pink-500 text-white rounded-lg overflow-hidden shadow-lg"
    type="button"
  >
    <div
      class="text-end p-2"
    >
      <svg
        class="FaArrowRight"
        color="white"
        fill="currentColor"
        height="22"
        stroke="currentColor"
        stroke-width="0"
        style="color: white;"
        viewBox="0 0 448 512"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        />
      </svg>
      <img
        alt="PayPal Holdings, Inc."
        class="company__logo p-3"
        src="https://financialmodelingprep.com/image-stock/PYPL.png"
      />
      <h2
        class="text-lg font-semibold"
      >
        PYPL
      </h2>
      <p>
        62.02
      </p>
    </div>
  </button>
</div>
`;
